src/Vulkan/RHI/VkCommandList.cpp

```diff
@@ -187,14 +190,76 @@ void VkCommandList::SetDescriptorPool(RHIDescriptorPool& descriptorPool, RHIReso
 
 void VkCommandList::SetInputAssembly(InputAssembly inputAssembly)
 {
-    VEX_NOT_YET_IMPLEMENTED();
+    commandBuffer->setPrimitiveRestartEnable(inputAssembly.primitiveRestartEnabled);
+    commandBuffer->setPrimitiveTopology(GraphicsPiplineUtils::InputTopologyToVkTopology(inputAssembly.topology));
 }
 
 void VkCommandList::ClearTexture(RHITexture& rhiTexture,
                                  const ResourceBinding& clearBinding,
                                  const TextureClearValue& clearValue)
 {
-    VEX_NOT_YET_IMPLEMENTED();
+    const TextureDescription& desc = rhiTexture.GetDescription();
+
+    u32 layerCount = clearBinding.sliceCount == 0 ? desc.depthOrArraySize : clearBinding.sliceCount;
+    u32 mipCount = clearBinding.mipCount == 0 ? desc.mips : clearBinding.mipCount;
+    //
+    // if (desc.usage & TextureUsage::RenderTarget && clearValue.flags & TextureClear::ClearColor)
+    // {
+    //     ::vk::ClearAttachment clearAttachment{
+    //         .aspectMask = static_cast<::vk::ImageAspectFlagBits>(clearValue.flags),
+    //         .colorAttachment = 0, // TODO: replace with correct attachment index?
+    //         .clearValue{
+    //             .color = ::vk::ClearColorValue{
+    //                 .float32 = clearValue.color
+    //             }
+    //         }
+    //     };
+    //
+    //     ::vk::ClearRect clearRect{
+    //         .rect = ::vk::Rect2D{
+    //             .offset = ::vk::Offset2D{ 0, 0 },
+    //             .extent = ::vk::Extent2D{ desc.width, desc.height }
+    //         },
+    //         .baseArrayLayer = clearBinding.startSlice,
+    //         .layerCount = layerCount,
+    //     };
+    //
+    //     commandBuffer->clearAttachments(1, &clearAttachment, 1, &clearRect);
+    // }
+    // else
+
+    if (desc.usage & TextureUsage::DepthStencil &&
```

Comment:
Sick! This code is way cleaner than the DX side thanks to ImageSubresourceRange!
Commented code is for clearing inside a clear rect? We probably should have a task to add ClearRects support on the API level (I've added a Trello).